tests/hf/test_integrals.py

```diff
@@ -282,6 +285,106 @@ def test_gradient(symbols, geometry, alpha, coeff):
+            2,
+            1,
+            np.array([1.5]),
+            np.array([0.0]),
+        ),
+        (  # trivial case, ref = 0.0 for e == 0 and t != 0
+            np.array([3.42525091]),
+            np.array([3.42525091]),
+            1,
+            0,
+            np.array([1.5]),
+            np.array([0.0]),
+        ),
+        (  # trivial case, ref = np.sqrt(np.pi / (alpha + beta))
+            np.array([3.42525091]),
+            np.array([3.42525091]),
+            0,
+            0,
+            np.array([1.5]),
+            np.array([0.677195]),
+        ),
+        (  # manually computed, ref = 1.0157925
+            np.array([3.42525091]),
+            np.array([3.42525091]),
+            0,
+            1,
+            np.array([1.5]),
+            np.array([1.0157925]),
+        ),
+    ],
+)
+def test_hermite_moment(alpha, beta, t, e, rc, ref):
+    r"""Test that _hermite_moment function returns correct values."""
+    assert np.allclose(_hermite_moment(alpha, beta, t, e, rc), ref)
+
+
+@pytest.mark.parametrize(
+    ("la", "lb", "ra", "rb", "alpha", "beta", "e", "rc", "ref"),
+    [
+        (  # manually computed, ref = 1.0157925
+            0,
+            0,
+            np.array([2.0]),
+            np.array([2.0]),
+            np.array([3.42525091]),
+            np.array([3.42525091]),
+            1,
+            np.array([1.5]),
+            np.array([1.0157925]),
+        ),
+    ],
+)
+def test_gaussian_moment(la, lb, ra, rb, alpha, beta, e, rc, ref):
+    r"""Test that gaussian_moment function returns correct values."""
+    assert np.allclose(gaussian_moment(la, lb, ra, rb, alpha, beta, e, rc), ref)
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "e", "idx", "ref"),
+    [
+        (
+            ["H", "Li"],
+            np.array([[0.0, 0.0, 0.0], [2.0, 0.0, 0.0]], requires_grad=False),
+            1,
+            0,  # 'x' component
+            3.12846324e-01,  # obtained from pyscf using mol.intor_symmetric("int1e_r")
+        ),
+        (
+            ["H", "Li"],
+            np.array([[0.5, 0.1, -0.2], [2.1, -0.3, 0.1]], requires_grad=True),
+            1,
+            0,  # 'x' component
+            4.82090830e-01,  # obtained from pyscf using mol.intor_symmetric("int1e_r")
+        ),
+        (
+            ["N", "N"],
+            np.array([[0.5, 0.1, -0.2], [2.1, -0.3, 0.1]], requires_grad=False),
+            1,
+            2,  # 'z' component
+            -4.70075530e-02,  # obtained from pyscf using mol.intor_symmetric("int1e_r")
+        ),
+    ],
+)
+def test_generate_moment(symbols, geometry, e, idx, ref):
+    r"""Test that generate_moment function returns a correct value for the moment integral."""
+    mol = Molecule(symbols, geometry)
+    basis_a = mol.basis_set[0]
+    basis_b = mol.basis_set[1]
+    args = [p for p in [geometry] if p.requires_grad]
+    s = generate_moment(basis_a, basis_b, e, idx)(*args)
+
+    assert np.allclose(s, ref)
+
```

Comment:
Should we be testing gradients of the integrals too? Can we even do it?

Reply:
Yes, added.